tiny_cnn/lossfunctions/loss_function.h

```diff
@@ -32,48 +32,67 @@ namespace tiny_cnn {
 // mean-squared-error loss function for regression
 class mse {
 public:
-    static float_t f(float_t y, float_t t) {
-        return (y - t) * (y - t) / 2;
+    static float_t f(const vec_t& y, const vec_t& t) {
+        assert(y.size() == t.size());
+        float_t d = 0.0;
+        for(unsigned int i = 0; i < y.size(); ++i)
+            d += (y[i] - t[i]) * (y[i] - t[i]);
+        return d/y.size();
     }
 
```

Comment:
@H4kor 
Could you tell me the reason why you change the API from ```float f(float,float)``` to ```float f(vec_t,vec_t)```, and ```float df(float,float)``` to ```vec_t df(vec_t,vec_t)```?
If we choose this API, using ```std::accumulate``` to avoid boilerplate for-loop seems better.